code/modules/mob/living/simple_animal/bot/floorbot.dm

```diff
@@ -90,6 +91,28 @@
 
 	return dat
 
+/mob/living/simple_animal/bot/floorbot/attackby(obj/item/W , mob/user, params)
+	if(istype(W, /obj/item/stack/tile/plasteel))
+		user << "<span class='notice'>The floorbot can produce normal tiles itself.</span>"
+		return
+	if(specialtiles != 0 && istype(W, /obj/item/stack/tile/))
+		var/obj/item/stack/tile/usedtile = W
+		if(usedtile != tiletype)
+			user << "<span class='warning'>Different custom tiles are already inside the floorbot.</span>"
+			return
+	if(istype(W, /obj/item/stack/tile/))
+		tiletype = W
```

Comment:
What ways would there be to keep the tile type known inside the bot? I just used the tiletype which I defined as var/obj/item/stack/tile/tiletype. If the game actually keeps checking the original item for reference even after putting the tiles in this should be changed.